Kernel/SortHelper.cpp

```diff
@@ -453,7 +459,7 @@ void SortHelper::collectVariableSortsIter(CollectTask task, DHMap<unsigned,TermL
 
             newTask.ts = binding;
             if (!isPredicate) {
-              newTask.contextSort = symbol->fnType()->result();
+              newTask.contextSort = SubstHelper::apply(type->result(),subst);
```

Comment:
During looking for an error, I fixed this part which wasn't correct I think. The question is whether this might have caused any issues, and if it should be also fixed for let tuples or should I just revert it.